elements/pfe-sass/mixins/_copy-mixins.scss

```diff
@@ -13,3 +13,107 @@
     text-decoration: underline;
     text-decoration-skip: ink;
 }
+
+
+/// ===========================================================================
+/// TYPOGRAPHY MIXIN for component use
+///
+/// This mixin allows you to request all typography styles with optional 
+/// hooks for local variables. Note that the default type is "title", because
+/// most text won't need typographical styles, it will inherit from the root.
+/// 
+/// USAGE:
+///
+/// .custom-heading {
+///   @include pfe-typography;
+/// }
+/// .text {
+///   @include pfe-typography($type: text);
+/// }
+/// .my-jump-links-component-class {
+///   @include pfe-typography("5xl", $LOCAL);
+/// }
+/// .custom-heading-6xl {
+///   @include pfe-typography($size: "6xl", $type: "mega-title");
+/// }
+/// .custom-heading-6xl-plus-local {
+///   @include pfe-typography($size: "6xl", $LOCAL: $LOCAL, $type: "mega-title");
+/// }
+/// .custom-heading-6xl-plus-local-and-region {
+///   @include pfe-typography($size: "6xl", $LOCAL: $LOCAL, $region: "heading", $type: "mega-title");
+/// }
+
+@mixin pfe-typography(
```

Comment:
Though your mixin is more explicit, there is some duplication here that could lead to having to update items in multiple places and potential missed changes.  If we collate into a single map then abstract the local calls, it might be easier to maintain:

```
@function str-to-camelcase($string) {
  $progress: $string;
  $result: "";
  $exclude: " ", "-", "–", "—", "_", ",", ";", ":", ".";
  
  @while str-length($progress) > 0  {
    $char: str-slice($progress, 1, 1);

    @if contain($exclude, $char) {
      $progress: capitalize(str-slice($progress, 2, 2)) + str-slice($progress, 3);
    }
    @else {
      $result: $result + $char;
      $progress: str-slice($progress, 2);
    }
 }
  
  @return $result;
}

@mixin pfe-typography(
    $size: "2xl", 
    $use-local: true, 
    $region: null,
    $type: "title") {
        // Font-family does not use local variables; only global theme values
        @if $type == "text" {
            font-family: pfe-var(font-family);
        }
        @else {
            font-family: pfe-var(font-family--heading);
        }
        
        // Set up the values to be used by the different types
        $font-stack: (
            // @TODO: Would be good to include a validate that the size is valid and exists in the var map
            font-size:       pfe-var(FontSize--#{to-string($size)}),
            line-height: (
                text:        pfe-var(line-height),
                mega-title:  pfe-var(line-height--heading),
                title:       pfe-var(line-height--heading)
            ),
            font-weight: (
                text:        pfe-var(font-weight--light),
                mega-title:  pfe-var(font-weight--light),
                title:       pfe-var(font-weight--normal)
            )
        );
        
        @each $prop, $set in $font-stack {
            $value: $set;
            @if type-of($set) == 'map' {
                $value: map-get($set, $type);
            }
            @if type-of($value) == 'string' {
                @if $use-local {
                    #{$prop}: pfe-local(str-to-camelcase($prop), $fallback: $value, $region: $region);
                }
                @else {
                    #{$prop}: #{$value};
                }
            }
        }
}
```

Reply:
It was just a back of the napkin mixin - I hadn't run it through a compiler or anything yet.  I did have a really hard time understanding the original mixin which is where my suggestion came from - trying to simplify and create an easier to read place to make changes if necessary.  I also kind of feel like the source of truth for these should not be sourced from inside the mixin because it makes the information less findable and harder to reuse.  Perhaps we can store the definitions in the main set of maps we use and pull them from there?